packages/ra-keycloak/src/authProvider.ts

```diff
@@ -188,27 +187,24 @@ export const keycloakAuthProvider = (
  * It seems the Keycloak init function may initially return before having authenticated the user.
  * To ensure we have the correct state, we need to wait for the onAuthSuccess event.
  */
-const isAuthenticated = (keycloakClient: Keycloak, timeout = 500) => {
+const isAuthenticated = (keycloakClient: Keycloak, timeout = 1000) => {
     return new Promise(resolve => {
+        // Resolve immediately if already authenticated
+        if (keycloakClient.authenticated && keycloakClient.token) {
+            keycloakClient.onAuthSuccess = null;
+            return resolve(true);
+        }
+
         const timeoutId = setTimeout(() => {
-            if (!keycloakClient.authenticated) {
-                keycloakClient.onAuthSuccess = null;
-                resolve(false);
-            }
+            keycloakClient.onAuthSuccess = null;
+            resolve(!!keycloakClient.authenticated);
```

Comment:
This is just a cleanup, to avoid creating the timeout and clearing it right away if the user is already authenticated